test/Dialect/FIRRTL/annotations.fir

```diff
@@ -1189,3 +1189,29 @@ circuit Top : %[[
     a is invalid
 
 ; CHECK-LABEL: firrtl.circuit
+
+
+; // -----
+;
+circuit memportAnno: %[[
+{
+ "class":"test",
+ "target":"~memportAnno|memportAnno/foo:Foo>memory.w"
+}
+]]
+  module memportAnno:
+    inst foo of Foo
+  module Foo:
+    mem memory:
+      data-type => UInt<8>
+      depth => 16
+      reader => r
+      writer => w
+      readwriter => rw
```

Comment:
This can be golfed down slightly by dropping the `writer` and `readwriter` ports.
```suggestion
```